api/src/test/java/bio/terra/tanagra/app/EntitiesFilterApiControllerTest.java

```diff
@@ -0,0 +1,47 @@
+package bio.terra.tanagra.app;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import bio.terra.tanagra.app.controller.EntitiesFiltersApiController;
+import bio.terra.tanagra.generated.model.ApiAttributeValue;
+import bio.terra.tanagra.generated.model.ApiAttributeVariable;
+import bio.terra.tanagra.generated.model.ApiBinaryFilter;
+import bio.terra.tanagra.generated.model.ApiBinaryFilterOperator;
+import bio.terra.tanagra.generated.model.ApiEntityFilter;
+import bio.terra.tanagra.generated.model.ApiFilter;
+import bio.terra.tanagra.generated.model.ApiSqlQuery;
+import bio.terra.tanagra.service.underlay.NauticalUnderlayUtils;
+import bio.terra.tanagra.testing.BaseSpringUnitTest;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.test.context.ActiveProfiles;
+
+@ActiveProfiles("nautical")
+public class EntitiesFilterApiControllerTest extends BaseSpringUnitTest {
+  @Autowired private EntitiesFiltersApiController apiController;
+
+  @Test
+  void generateSqlQuery() {
+    ApiEntityFilter apiEntityFilter =
+        new ApiEntityFilter()
```

Comment:
Side thought: I wonder how useful it would be to have a query string rendering of the entity filter, either as an input or simply as a debugging tool.

Reply:
I'm not sure what you mean. Could you say more?
I think eventually we want to accept/export in APIs a JSON ApiEntityFilter + underlay + entity (the underlay and entity being path parameters today). Is that what you mean?

Reply:
No, sorry - I wasn't very clear. I was talking about being able to parse or generate a SQL-like query string for a Tanagra filter, i.e. probably something that probably looks similar to JQL.

For example, in this case if I could render this SQL request as
```
SELECT s.id FROM Sailors s WHERE s.rating = 42;
```

Reply:
Got it, yeah. We could make a DSL, but I'd have a hard time prioritizing it unless it's very helpful for clients.